src/api.rs

```diff
@@ -56,8 +57,29 @@ impl Api {
         Ok(Json(plan))
     }
 
+    /// Retrieve a Data Product
+    #[oai(path = "/data_product/:dataset_id/:data_product_id", method = "get", tag = Tag::DataProduct)]
+    async fn data_product_get(
```

Comment:
@coderabbitai Please add a unit test where this endpoint can read a data product record from the DB, and a unit test where there data product does not exists.